Sources/protoc-gen-swift/GeneratorOptions.swift

```diff
@@ -17,9 +17,34 @@ class GeneratorOptions {
     case dropPath = "DropPath"
   }
 
-  enum Visibility : String {
-    case `internal` = "Internal"
-    case `public` = "Public"
+  enum Visibility {
+    case `internal`
+    case `public`
+    case spi(String)
+
+    init?(rawValue: String) {
+      if rawValue == "Internal" {
+        self = .internal
+      } else if rawValue == "Public" {
+        self = .public
+      } else if rawValue.hasPrefix("_SPI(") && rawValue.hasSuffix(")") {
```

Comment:
I would recommend against using parens (shell special characters) here. It is unpleasant when you try to cut and past commands, or everyone needs to deal with escape characters, etc. I wonder if a separate parameter is in order, even though it's not orthogonal.

Reply:
Valid point, since this should be rarely used I am unsure if making this more complicated is worth it.
I went with the parenthesis to keep it in line with how the actual attribute works, but we can also just switch it to something like SPI/Foo